drivers/SmartThings/zigbee-motion-sensor/src/aqara/aqara_utils.lua

```diff
@@ -24,6 +24,10 @@ local function motion_detected(device)
   end
 
   local detect_duration = device:get_field(PREF_FREQUENCY_KEY) or PREF_FREQUENCY_VALUE_DEFAULT
+  if type(detect_duration) == "string"  then
+    detect_duration = tonumber(detect_duration)
```

Comment:
`tonumber` can return `nil` too. If we are having issue with protocol compliance, maybe we should handle that case too, in case it sends a string that can not be converted to a number for some reason.

Reply:
In case tonumber return nil, we modify line 28 as below.
```suggestion
    detect_duration = tonumber(detect_duration) or PREF_FREQUENCY_VALUE_DEFAULT
```

Reply:
This suggestion looks good

Reply:
I also agree with the above suggestion.